app/controllers/wpcc/your_results_controller.rb

```diff
@@ -1,17 +1,44 @@
 module Wpcc
   class YourResultsController < EngineController
-    def index
-      @schedule = Wpcc::Presenter.new(schedule, view_context: view_context)
-      @message_presenter = message_presenter
+    def schedule
+      calendar.schedule.map do |period_contribution|
+        Wpcc::PeriodContributionPresenter.new(
+          period_contribution,
+          view_context: view_context
+        )
+      end
     end
+    helper_method :schedule
 
-    private
+    def period_percents
+      calendar.period_percents.map do |period_percent|
+        Wpcc::PeriodContributionCalculatorPresenter.new(
+          period_percent,
+          view_context: view_context
+        )
+      end
+    end
+    helper_method :period_percents
 
-    def schedule
-      present(
-        Wpcc::ContributionsCalendar.new(contributions_params).schedule
+    def message_presenter
+      Wpcc::MessagePresenter.new(
+        salary_message,
+        view_context: view_context
       )
     end
+    helper_method :message_presenter
+
+    def salary_frequency
+      @salary_frequency =
+        params[:salary_frequency] || session[:salary_frequency]
```

Comment:
I may be jumping ahead in this review.. but does the session also get updated if new params are passed in - and if so, is this before or after this action?